### DISCUSSION TITLE
Error on customization "copy link" button

### DISCUSSION THREAD

**Discussion by rockjaylee** [rejetto/hfs#785] [Q&A] [answered]
I followed the steps [here](https://github.com/rejetto/hfs/wiki/Customization#add-a-copy-link-to-file-menu) to add "copy link" button to the file menu.
And when I click on the button it says: ERROR!  Cannot read properties of undefined (reading 'writeText')
![Snipaste_2024-10-21_15-58-10](https://github.com/user-attachments/assets/f488d9d6-608c-4b0c-b6c7-cb0f0a608640)

hfs version: 0.54.0

**Comment by SanokKule** [comment]
It won't work if you don't have https enabled

**Reply by SanokKule** [reply, marked as answer]
https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText#security_considerations

**Reply by rockjaylee** [reply]
thanks

**Reply by SanokKule** [reply]
Did you check the new code from the wiki?
It should work even without https

**Reply by rockjaylee** [reply]
yes it worked on http, thanks so much!

**Comment by rejetto** [comment]
thanks Sanok!
i wrote that now in the wiki, but it would be better to modify the code in a way that it always works.

**Comment by rejetto** [comment]
ok, i changed the code for better compatibility

**Comment by SimonChiu930** [comment]
add "copy link" button to the file menu, when click menu back, will duplicate "Copy Link"
![image](https://github.com/user-attachments/assets/10465ec6-cff8-447f-8a63-db7802a07d31)

but refresh the page will normal

**Reply by rejetto** [reply]
i don't understand how to reproduce this problem, sorry.
i tried, but didn't succeed.
can you make a video? or say exactly the sequence of clicks to do, without omitting anything

**Comment by SimonChiu930** [comment]
I just discovered that after logging in, "Copy Link" will be duplicate.
Press F5 reload and it will return to normal.

https://github.com/user-attachments/assets/cf9b7a92-3afb-465f-bef4-6e8ef0635c35

**Reply by rejetto** [reply]
ok, this will be fixed in next release. THanks for your collaboration

**Reply by SimonChiu930** [reply]
Thanks

**Comment by rejetto** [comment]
i tried with chrome firefox and safari, both on mac and windows, and couldn't reproduce the problem.
what browser are you using?

**Comment by SimonChiu930** [comment]
I tried using Edge and Chrome, including incognito, and the same thing happened. I didn't think much of it when I started using it. I only found out about it after a colleague consulted me one day, but it didn't have much impact.

**Comment by rejetto** [comment]
it must be something in your hfs configuration. If you show me your config i can take a look
admin-panel > options > config file > copy without passwords
here or a@rejetto.com

**Comment by killersolyom** [comment]
It reproduces for me as well (the duplicated copy button bug). I found a workaround that seems to work but I'm not sure it's a good solution.
Before Header:

```
<script>

    function attachListeners() {
        if (!window.fileMenuListenerAttached) {  
            // Copy url    
            HFS.onEvent('fileMenu', ({ entry }) => ({
                label: "Copy Link",
                async onClick() {
                await navigator.clipboard.writeText(location + entry.n);
            }}));
            
            // Mark the listener as attached
            window.fileMenuListenerAttached = true;
          }
    }
    
    attachListeners();

</script>
```

**Reply by rejetto** [reply]
i guess it's ok, but you won't need that in next release
https://github.com/rejetto/hfs/discussions/785#discussioncomment-11274209
